scalene/scalene_profiler.py

```diff
@@ -1189,20 +1187,11 @@ def should_trace(filename: str) -> bool:
         if "scalene/scalene" in filename:
             # Don't profile the profiler.
             return False
-        found_in_profile_only = False
-        for prof in profile_only_list:
-            if prof in filename:
-                found_in_profile_only = True
-                break
-
+        found_in_profile_only = any(prof in filename for prof in profile_only_list)
         if not found_in_profile_only:
             return False
         if Scalene.__args.profile_all:
             # Profile everything else, except for "only" choices.
-            found_in_profile_only = False
-            for prof in profile_only_list:
-                if prof in filename:
-                    return True
             if profile_only_list and not found_in_profile_only:
```

Comment:
As it is now, `not found_in_profile_only` must be `True` at this point.

Reply:
Good point, fixed.